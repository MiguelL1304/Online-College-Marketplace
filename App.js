import * as React from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

import Login from "./app/screens/Login";
import Listing from "./app/screens/Listing";
import SignUp from "./app/screens/SignUp";
import Profile from "./app/screens/Profile";


const Stack = createNativeStackNavigator();

export default function App() {
  

  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen 
          name="Login" 
          component={Login} 
          options={{
            headerShown: false,
          }}
        />
        <Stack.Screen 
          name="SignUp" 
          component={SignUp} 
          options={{
            headerTitle: '',
          }}
        />
        <Stack.Screen 
          name="Profile" 
          component={Profile} 
          options={{
            headerShown: false,
          }}
        />
        <Stack.Screen 
          name="Listing" 
          component={Listing} 
        />
      
      </Stack.Navigator>
    </NavigationContainer>
    
    //<View style={styles.container}>
      //<Text>Open up App.js to start working on your app!!!</Text>
      //<StatusBar style="auto" />
    //</View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
