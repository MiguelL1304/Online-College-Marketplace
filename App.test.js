import React from 'react';
import renderer from 'react-test-renderer';
import AsyncStorage from "@react-native-async-storage/async-storage";

import App from '../UniSwap/App';

// Mock AsyncStorage functions
jest.mock("@react-native-async-storage/async-storage", () => ({
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
}));

describe('<App />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});