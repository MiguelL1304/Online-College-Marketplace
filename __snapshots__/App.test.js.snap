// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              [
                {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <View
              style={
                {
                  "width": "80%",
                }
              }
            >
              <TextInput
                onChangeText={[Function]}
                placeholder="Email"
                style={
                  {
                    "backgroundColor": "#d4e9fa",
                    "borderRadius": 10,
                    "marginTop": 5,
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                  }
                }
                value=""
              />
              <TextInput
                onChangeText={[Function]}
                placeholder="Password"
                secureTextEntry={true}
                style={
                  {
                    "backgroundColor": "#d4e9fa",
                    "borderRadius": 10,
                    "marginTop": 5,
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                  }
                }
                value=""
              />
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginTop": 40,
                  "width": "60%",
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#3f9eeb",
                    "borderRadius": 10,
                    "opacity": 1,
                    "padding": 15,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 16,
                      "fontWeight": "700",
                    }
                  }
                >
                  Login
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "#3f9eeb",
                    "borderRadius": 10,
                    "borderWidth": 2,
                    "marginTop": 5,
                    "opacity": 1,
                    "padding": 15,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#3f9eeb",
                      "fontSize": 16,
                      "fontWeight": "700",
                    }
                  }
                >
                  Register
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
